src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -1454,40 +1461,61 @@ cu_done(struct sched_cmd *cmd)
 	return false;
 }
 
-inline int
+//Copy soft kernel return code to execBO
+static inline void
+copy_sk_return(struct sched_cmd *cmd)
+{
+	struct drm_zocl_dev *zdev = cmd->ddev->dev_private;
+	struct soft_krnl *sk = zdev->soft_kernel;
+	struct soft_cu *scu;
+	uint32_t *cu_regfile;
+	struct ert_start_kernel_cmd *skc;
+
+	skc = (struct ert_start_kernel_cmd *)cmd->packet;
+	scu = sk->sk_cu[cmd->cu_idx];
+	cu_regfile = scu->sc_vregs;
+
+        skc->data[skc->extra_cu_masks + 1] = cu_regfile[1];
```

Comment:
Also, I suggest we add some comment and document it later that the CU (include soft CU and hard CU) return code is put into a fixed place of ert_start_kernel_cmd structure. 
There seems to be an issue for the place at pkt->data[pkt->extra_cu_maskes + 1] when pkt->count = (0 or 1). It overlaps with the command state. see ert_start_kernel_timestamps

Reply:
updated. Please check again